whois-nrtm4/src/main/java/net/ripe/db/nrtm4/dao/SnapshotFileRepository.java

```diff
@@ -42,6 +42,20 @@ INSERT INTO snapshot_file (version_id, name, hash, payload)
             payload);
     }
 
+    public Optional<SnapshotFile> getByVersionID(final long versionID) {
+        final String sql = """
+            SELECT sf.id, sf.version_id, sf.name, sf.hash
+            FROM snapshot_file sf
+            WHERE sf.version_id = ?
+            """;
+        try {
+            return Optional.ofNullable(jdbcTemplate.queryForObject(sql, rowMapper, versionID));
```

Comment:
Why use `ofNullable` rather than `of`? Can the value ever be `null`?

Reply:
technically, yes (and intellij warns about it) but in practice an exception will be thrown and empty() will be returned. so it's really to stop intellij nagging -- it makes no difference for our code.

Reply:
OK